Fix #4047: Add "aria-required" attribute and refactored ARIA attributes.

melloware opened the pull request:
OK this was a bigger change than I thought but every time I stepped back and looked at it I could make the code cleaner and cleaner.  What this PR does...

1. Adds "aria-required" to any UIInput component that is marked required=true.
2. Cleans up ARIA handling and "disabled" and "readonly" attribute handling to prevent cut and paste code.
3. Add ARIA handling to components that were missing it like Signature, Rating, SelectOneMenu etc etc etc.

Note: I know there is client side JS for "PrimeFaces.skinInput()" which adds aria-disabled and aria-readonly but that is A) only for input components and B) rquires developers to know to call it in every  place.  This PR ensures those attributes are set on any Input component that doesn't even need to call skinInput like Signature and Rating for example but renders the proper ARIA attributes.

@Rapster @mertsincan @tandraschko Please review.  This change seems more invasive than I think it is but cleans up the ARIA code immensely IMHO.

melloware committed:
Fix #4034: Excel streaming API track column sizes.

melloware committed:
Fix #3791: SelectOne ClassCastException for editable="true".

wsaca committed:
Align panel when it's filtered.

melloware committed:
Fix #4043: Update dialogs to latest ARIA standards.

mertsincan committed:
Fixed #3269

mertsincan committed:
Refactor #3269

melloware committed:
Fix #4047: Add "aria-required" attribute and refactored ARIA attributes to reduce cut and paste code.

Rapster commented:
The code might be invasive, but as you said it gives a better readabiltity, less duplication and it's optimized for JIT. So totally worth it :wink: Basically, I like it :+1: Great job @melloware!

melloware committed:
Fix #4049: Refactor calls to renderValidationMetadata.

tandraschko commented:
I wonder If the isreadonly and isdisabled method should only allow uiinput?
Other components should not have this attributes.

Rapster commented:
Agreed

melloware commented:
Agreed will update.

melloware committed:
Fix #4047: Changed UIComponent to UIInput for disabled and readonly as no other components should have these properties.

melloware commented:
OK I have changed UIComponent to UIINput for shouldDecode(), isDisabled() and isReadOnly() methods.

I don't know why this still says I have conflicts when I rebased my branch and resolved them all?

tandraschko commented:
Maybe related because i unintenionally used rebase instead merging on my phone to apply PRs?

melloware committed:
Fix #4047: Change renderRequired to renderAccessibility for some components.

tandraschko commented:
I really Like the changes!
I would merge it If there are no conflicts.

Maybe its because of the Fileupload stuff? Its displayed as changes but its the same as trunk

melloware commented:
Maybe I rebased my branch incorrectly?  I did "git rebase origin/master" and then resolved the conflicts and updated my branch.  IS that not the right way to fix the conflicts?  I am a NOOB when it comes to this type of stuff with Git.

melloware commented:
You know what will be easier I will just make a new PR with all the changes for review and close this one when I am done.  I think it will be easier than figuring out what mess I made of this branch. 😄
